### ISSUE TITLE
Design Update

### ISSUE THREAD

**Issue by timkpaine** [timkpaine/jupyter-fs#9]
**tl;dr: These aren't the final designs, they're just some things we tried along the way.** We're doing more testing and iterating before we arrive at a final design.

---

We’ve been continuing our research on file system design, and while we don’t have a final design ready to share, we *have* arrived at a direction to take with the design. More to come on that.
For now, in the interest of openness, we’d like to share some of our progress, and document some of the steps we’ve taken along the way.
There is, of course, our [previous](https://github.com/jpmorganchase/jupyter-fs/issues/36) [research](https://github.com/jupyter/design/issues/63) that we’ve shared. We also shared some of our [early wireframes](https://github.com/jpmorganchase/jupyter-fs/issues/39), as we were just beginning to put pen to paper. Since then, we’ve had the opportunity to work more with the Jovyan community, and learned a lot about the specifics of how Jupyter users expect a file system to work, and what they need from their file browser going forward.
Using the data we had gathered from our interviews and heuristic evaluation, we started down two different design tracks, allowing ourselves to explore these alternatives so we wouldn’t get ourselves stuck in a rut, fixated on a single concept early on.

## Concept 1: "Mac Finder Lite"
The first of these, "Mac Finder Lite," as we called it, would move the file browser from the left-hand sidebar into the main work area. This concept tested well with users who are newer to Jupyter Lab. However, it requires a big mental model shift for the more experienced users — and also represents a *significant* amount of development effort. Here’s some of those wireframes:
![01](https://user-images.githubusercontent.com/2984983/87612811-70fef900-c6c0-11ea-8c2b-083f22777642.jpg)
![02](https://user-images.githubusercontent.com/2984983/87612817-72c8bc80-c6c0-11ea-9f8c-40fb92ca7b52.jpg)
![03](https://user-images.githubusercontent.com/2984983/87612819-73f9e980-c6c0-11ea-9132-fc17f29282f1.jpg)
![04](https://user-images.githubusercontent.com/2984983/87612820-74928000-c6c0-11ea-8653-cbb415f9ca7f.jpg)
![05](https://user-images.githubusercontent.com/2984983/87612821-752b1680-c6c0-11ea-9244-0d7f960abb8c.jpg)
![06](https://user-images.githubusercontent.com/2984983/87612823-752b1680-c6c0-11ea-966f-9699e9561727.jpg)
![07](https://user-images.githubusercontent.com/2984983/87612824-75c3ad00-c6c0-11ea-862c-24c132b261d5.jpg)

## Concept 2: "Launcher Add-On"
The second, "Launcher Add-On," would keep the file browser in the sidebar, but offer the ability to open multiple browsers at once. This concept was much more popular with the more experienced users of Jupyter Lab, and is a better fit for the existing mental model, while still gaining some of the more powerful features many users requested in interviews. Take a look:
![11](https://user-images.githubusercontent.com/2984983/87612828-7a886100-c6c0-11ea-92e5-c3e0589f8c2e.jpg)
![12](https://user-images.githubusercontent.com/2984983/87612830-7b20f780-c6c0-11ea-86cd-5c5447bb0167.jpg)
![13](https://user-images.githubusercontent.com/2984983/87612831-7bb98e00-c6c0-11ea-850a-e33d9ca2b011.jpg)
![14](https://user-images.githubusercontent.com/2984983/87612832-7bb98e00-c6c0-11ea-8f5e-605b4bcf7960.jpg)

## Moving Forward
Having done our exploratory designs, we’re looking forward to the direction to take with the final design. "Launcher Add-On" is a better fit for Jupyter Lab, and will be the primary focus of our future designs; however, we did learn some valuable things from "Mac Finder Lite," and will be offering those findings as recommended next-steps to the Jupyter team.
As we move forward, we’re redoubling our focus on two key problems, which we heard from *every Jupyter user* that we talked with: finding the right files, and organizing them once you’ve found them. These aren’t the *only* problems we want to address, but they’re the two *universal* problems, and so they’re our primary focus.
We’re going to develop a consolidated design and take it through a few rounds of user testing. As of this writing, we’ve got a few sketches, which we’ll be developing into wireframes, and we’ll share those wireframes when appropriate. (We’d share some sketches, but speaking for myself, my handwriting is so atrocious that they’re not worth the effort to scan and upload.)
Thank you for reading, and we hope that this update gives you a good view into our design process. We do, of course, welcome feedback on these design artifacts, but keep in mind that **these are not the final design.** Don’t get too attached, and feel free to point out any issues - we’d hate to bring something forward because we didn’t realize there was an issue.
Thanks,
The UCI team: @jabumeri, @JDLEarley, @emilythefan, @Nastraughn, and myself

---

xref: https://github.com/jpmorganchase/jupyter-fs/issues/45
